osu.Game/Screens/SelectV2/WedgeDifficultyDisplay.cs

```diff
@@ -0,0 +1,375 @@
+                                                               .ToList();
+
+            updateDifficultyStatistics();
+        }
+
+        private void updateDifficultyStatistics() => Scheduler.AddOnce(() =>
+        {
+            BeatmapDifficulty baseDifficulty = beatmap.Value.BeatmapInfo.Difficulty;
+            BeatmapDifficulty originalDifficulty = new BeatmapDifficulty(baseDifficulty);
+
+            foreach (var mod in mods.Value.OfType<IApplicableToDifficulty>())
+                mod.ApplyToDifficulty(originalDifficulty);
+
+            var rateAdjustedDifficulty = originalDifficulty;
+
+            if (ruleset.Value != null)
+            {
+                double rate = ModUtils.CalculateRateWithMods(mods.Value);
+
+                rateAdjustedDifficulty = ruleset.Value.CreateInstance().GetRateAdjustedDisplayDifficulty(originalDifficulty, rate);
+                difficultyStatisticsDisplay.TooltipContent = new AdjustedAttributesTooltip.Data(originalDifficulty, rateAdjustedDifficulty);
+            }
+
+            WedgeStatisticDifficulty.Data firstStatistic;
+
+            switch (ruleset.Value?.OnlineID)
+            {
+                case 3:
+                    // Account for mania differences locally for now.
+                    // Eventually this should be handled in a more modular way, allowing rulesets to return arbitrary difficulty attributes.
+                    ILegacyRuleset legacyRuleset = (ILegacyRuleset)ruleset.Value.CreateInstance();
+
+                    // For the time being, the key count is static no matter what, because:
+                    // a) The method doesn't have knowledge of the active keymods. Doing so may require considerations for filtering.
+                    // b) Using the difficulty adjustment mod to adjust OD doesn't have an effect on conversion.
+                    int keyCount = legacyRuleset.GetKeyCount(beatmap.Value.BeatmapInfo, mods.Value);
+
+                    firstStatistic = new WedgeStatisticDifficulty.Data(BeatmapsetsStrings.ShowStatsCsMania, keyCount, keyCount, 10);
+                    break;
+
+                default:
+                    firstStatistic = new WedgeStatisticDifficulty.Data(BeatmapsetsStrings.ShowStatsCs, originalDifficulty.CircleSize, rateAdjustedDifficulty.CircleSize, 10);
+                    break;
+            }
+
+            difficultyStatisticsDisplay.Statistics = new[]
+            {
+                firstStatistic,
+                new WedgeStatisticDifficulty.Data(BeatmapsetsStrings.ShowStatsAccuracy, originalDifficulty.OverallDifficulty, rateAdjustedDifficulty.OverallDifficulty, 10),
+                new WedgeStatisticDifficulty.Data(BeatmapsetsStrings.ShowStatsDrain, originalDifficulty.DrainRate, rateAdjustedDifficulty.DrainRate, 10),
+                new WedgeStatisticDifficulty.Data(BeatmapsetsStrings.ShowStatsAr, originalDifficulty.ApproachRate, rateAdjustedDifficulty.ApproachRate, 10),
```

Comment:
Did you mean this? Else hard rock / easy has no visual change.

```suggestion
                    firstStatistic = new WedgeStatisticDifficulty.Data(BeatmapsetsStrings.ShowStatsCs, baseDifficulty.CircleSize, rateAdjustedDifficulty.CircleSize, 10);
                    break;
            }

            difficultyStatisticsDisplay.Statistics = new[]
            {
                firstStatistic,
                new WedgeStatisticDifficulty.Data(BeatmapsetsStrings.ShowStatsAccuracy, baseDifficulty.OverallDifficulty, rateAdjustedDifficulty.OverallDifficulty, 10),
                new WedgeStatisticDifficulty.Data(BeatmapsetsStrings.ShowStatsDrain, baseDifficulty.DrainRate, rateAdjustedDifficulty.DrainRate, 10),
                new WedgeStatisticDifficulty.Data(BeatmapsetsStrings.ShowStatsAr, baseDifficulty.ApproachRate, rateAdjustedDifficulty.ApproachRate, 10),
```

Also do we want rate changes / HR or EZ or DA to have the same display? Rate changes don't really affect the number value, so just pointing this out if it's a problem.

Reply:
That matches how lazer currently behaves no? Not looking to change behaviour here.

Reply:
Was that in response to my other question at the bottom? If so, yes, rate changes / HR, etc. display the same right now. The addition of arrows made me want to compare to stable's way of displaying it differently:
- up carets with no value change / color when DT is enabled
- red color and value change when HR is enabled

Reply:
That feels arbitrary to be honest, there's no real reason not to show the change in the bar when DT is selected. I would rather keep this as-is and let it be reiterated later if deemed better.

Reply:
> there's no real reason not to show the change in the bar when DT is selected

Not what I'm saying. Anyway this is just a little detail not worth discussing further.